src/main/java/reposense/parser/RepoConfigCsvParser.java

```diff
@@ -42,6 +44,21 @@ public RepoConfigCsvParser(Path csvFilePath) throws IOException {
         };
     }
 
+    /**
+     * Parses the csv file associated with this instance of the {@code CsvParser} and returns a {@code List}
+     * containing the RepoConfigurations in this file.
+     *
+     * @throws IOException if there are 0 valid records in the csv, or if there are errors accessing the file.
+     */
+    @Override
+    public List<RepoConfiguration> parse() throws IOException {
+        List<RepoConfiguration> repoConfigs = super.parse();
+        if (repoConfigs.isEmpty()) {
+            throw new IOException(MESSAGE_ZERO_REPO_CONFIGS);
+        }
+        return repoConfigs;
```

Comment:
can't such a solution be introduced in the parent method so that it will apply to author config as well?

Reply:
This overriding version of `parse` will terminate RepoSense execution if there are no valid records in the CSV.

Is this appropriate for the author config? I felt that we **should not** terminate RepoSense if there are no valid records in the author config, since author config is an optional file.

Reply:
Yes but the work on this PR doesn't seems to be addressing the original problem stated on #872 
So perhaps you should first address the problem and this PR can come after that.

Reply:
@eugenepeh, sorry for the confusion.

The original problem described in #872 is that RepoSense terminates with an unhelpful error when the user does not provide the correct number of columns in a CSV.

This PR solves the problem that RepoSense terminates with an unhelpful error when there are zero valid configs in the `repo-config.csv`.

I should have created a separate issue for this PR. I will do so now.
My apologies for the error.

Reply:
For the original problem given in #872, I plan to tackle it after PR #903 is merged, as my envisioned solution relies on the `getHeader()` method introduced in PR #903.